src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -218,7 +218,118 @@ private static unsafe int IndexOfVectorized(ref byte searchSpace, byte value, in
         exitFixed:;
                 return offset;
             }
+        }
+
+        public static bool TryIndicesOf(this Span<byte> buffer, byte value, Span<int> indices, out int numberOfIndices)
```

Comment:
Yeah, I meant to return false when search terminated due to running out of buffer space for the indices.

That would make the logic as follows (which is same as we do for encoders):
- If result == true, search completed entire buffer and indices contains all hits.
- If result == false and numberOfIndices == indices.Length, we found some hits and returned them to you, but ran out of room in the destination buffer. There may or may not be more hits further on in your data.
- If result == false and numberOfIndices returns 0, there were no hits (or you passed us an empty destination buffer so that's on you).